File: be/src/statistics/dto/timeline-statistics.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class TimelineStatisticsDto {
  @ApiProperty({ example: '2024-01-15', description: 'Ngày thống kê' })
  date: string;

  @ApiProperty({ example: 5, description: 'Số nhiệm vụ được tạo' })
  createdTasks: number;

  @ApiProperty({ example: 3, description: 'Số nhiệm vụ hoàn thành' })
  completedTasks: number;

  @ApiProperty({ example: 2, description: 'Số nhiệm vụ hoàn thành đúng hạn' })
  onTimeTasks: number;

  @ApiProperty({ example: 1, description: 'Số nhiệm vụ hoàn thành trễ hạn' })
  lateTasks: number;
}

File: backend-java/src/main/java/com/qlda/backend/statistics/dto/TimelineStatisticsDto.java
package com.qlda.backend.statistics.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TimelineStatisticsDto {
    private String date;
    private Integer createdTasks;
    private Integer completedTasks;
    private Integer onTimeTasks;
    private Integer lateTasks;
}

